tools/Google.Cloud.Tools.SourceManipulation/SourceFile.cs

```diff
@@ -0,0 +1,293 @@
+                .SelectMany(typeDeclaration => typeDeclaration.ChildNodes().OfType<MethodDeclarationSyntax>())
+                .ToList();
+            var matchingMethods = methods
+                .Where(method => method.Identifier.Text == methodName)
+                .Where(method => method.ParameterList.Parameters.Count == parameterTypes.Length &&
+                    method.ParameterList.Parameters.Zip(parameterTypes, (actual, expected) => (actual, expected))
+                        .All(pair => ParameterMatches(pair.actual, pair.expected)))
+                .ToList();
+            if (matchingMethods.Count != 1)
+            {
+                string friendlyName = $"{typeName}.{methodName}({string.Join(", ", parameterTypes)})";
+                throw new Exception($"Expected to find one method matching {friendlyName}; found {matchingMethods.Count}");
+            }
+            return WithNodeRemoved(matchingMethods[0]);
+
+            bool ParameterMatches(ParameterSyntax syntax, string expectedType)
+            {
+                switch (syntax.Type)
+                {
+                    case AliasQualifiedNameSyntax alias:
+                        return alias.Name.Identifier.Text == expectedType;
+                    case IdentifierNameSyntax identifier:
+                        return identifier.Identifier.Text == expectedType;
+                    case PredefinedTypeSyntax predefined:
+                        return predefined.Keyword.Text == expectedType;
+                    case QualifiedNameSyntax qualified:
+                        // This is pretty permissive, but unlikely to cause problems
+                        return qualified.Right.Identifier.Text == expectedType;
+                    default:
+                        throw new Exception($"Unexpected parameter type syntax: {syntax.Type.Kind()}");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Removes the specified property.
+        /// </summary>
+        /// <returns>The modified the source file.</returns>
+        public SourceFile RemoveProperty(string typeName, string propertyName)
+        {
+            var matchingProperties = _root.DescendantNodes().OfType<TypeDeclarationSyntax>()
+                .Where(typeDeclaration => typeDeclaration.Identifier.Text == typeName)
+                .SelectMany(typeDeclaration => typeDeclaration.ChildNodes().OfType<PropertyDeclarationSyntax>())
+                .Where(property => property.Identifier.Text == propertyName)
+                .ToList();
+            if (matchingProperties.Count != 1)
+            {
+                throw new Exception($"Expected to find one property matching {typeName}.{propertyName}; found {matchingProperties.Count}");
+            }
+            return WithNodeRemoved(matchingProperties[0]);
+        }
+
+        /// <summary>
+        /// Removes the specified field.
+        /// </summary>
+        /// <returns>The same object, for chaining.</returns>
```

Comment:
nit: This is not quite correct; it's not returning the same object, it's returning a new `SourceFile` with the requested changes applied.

Reply:
Doh - this was due to a design change. I failed to update all the comments :(